server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java

```diff
@@ -0,0 +1,310 @@
+            } else if (osCapability.getArchitecture().get().equals(OsCapability.Architecture.x86_32)) {
+                describeImagesRequest.withFilters(new Filter("architecture").withValues("i386"));
+            }
+        } else {
+            //defaulting to 64 bit architecture
+            describeImagesRequest.withFilters(new Filter("architecture").withValues("x86_64"));
+        }
+        String imageId;
+        try {
+            DescribeImagesResult describeImagesResult = ec2.describeImages(describeImagesRequest);
+            Integer numReceivedImages = describeImagesResult.getImages().size();
+            logger.debug("Got " + numReceivedImages + " images from aws");
+            if (numReceivedImages > 0) {
+                DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+                Map<Date, Image> creationDateMap = new HashMap<>();
+                for (Image image : describeImagesResult.getImages()) {
+                    try {
+                        Date date = dateFormat.parse(image.getCreationDate());
+                        creationDateMap.put(date, image);
+                    } catch (ParseException pE) {
+                        logger.error("Error parsing dateformat");
+                        pE.printStackTrace();
+                    }
+                }
+                Image latest = creationDateMap.get(Collections.max(creationDateMap.keySet()));
+                logger.debug("Latest image received: " + latest.toString());
+                imageId = latest.getImageId();
+            } else {
+                logger.warn("No images received defaulting to old ubuntu 16.04 image");
+                imageId = "ami-0def3275";
+                //TODO maybe not defaulting but throwing a transformation failed exception?
+            }
+        } catch (SdkClientException se) {
+            logger.error("Cannot connect to AWS to request image Ids, defaulting to old ubuntu 16.04 image");
+            imageId = "ami-0def3275";
+            //TODO maybe not defaulting but throwing a transformation failed exception?
+        }
+        logger.debug("ImageId is: " + imageId);
+        return imageId;
+    }
+
+    /**
+     Finds the best InstanceType based on the values contained in the ComputeCapability.
+     If necessary the values are scaled upwards till they meet the requirement.
+
+     @param computeCapability The ComputeCapability to map.
+     @param distinction       A distinction string. Can either be "EC2" or "RDS".
+     @return A valid InstanceType / InstanceClass string.
+     @throws TransformationFailureException Gets thrown if the values numCpus and memSize are too big and there is no
+     valid InstanceType.
+     */
+    public String mapComputeCapabilityToInstanceType(ComputeCapability computeCapability, String distinction) throws
+        TransformationFailureException {
+        //TODO what to do with disk size?
```

Comment:
Is this maybe possible with attached volumes? Can you change the root volume of the EC2 instance?
Added an issue for this: #423